File: LeetCode/Easy/1331-rank-transform-of-an-array/1331-rank-transform-of-an-array.java
class Solution {
    public int[] arrayRankTransform(int[] arr) {
        int[] temp = arr.clone();
        Arrays.sort(temp);
        HashMap<Integer, Integer> map = new HashMap();
        
        int i = 0;
        int j = 1;
        while (i < arr.length){
            map.put(temp[i++], j++);
            while (i < arr.length && temp[i] == temp[i-1]) ++i;
        }
        
        for (int k = 0; k < arr.length; ++k) arr[k] = map.get(arr[k]);
        
        return arr;
    }
}

File: LeetCode/Easy/1331-rank-transform-of-an-array/1331-rank-transform-of-an-array.cpp
class Solution {
public:
    vector<int> arrayRankTransform(vector<int>& arr) {
        vector<int> temp = arr;
        sort(temp.begin(), temp.end());
        unordered_map<int, int> map;
        
        int i = 0;
        int j = 1;
        
        while (i < arr.size()){
            map[temp[i++]] = j++;
            while (i < arr.size() && temp[i] == temp[i-1]) ++i;
        }
        
        for (int k = 0; k < arr.size(); ++k) arr[k] = map[arr[k]];
        
        return arr;
    }
};